<launch>
    <arg name="record" default="false"/>

    <include file="$(find turtlebot_gazebo)/launch/turtlebot_world.launch">
        <arg name="world_file" value="$(find pytheas)/world/finalProject.world"/>
    </include>

    <node
        pkg="pytheas"
        name="FinalProject"
        type="turtlebotFinal"
	output="screen" 
	launch-prefix="gnome-terminal --command"
    />

    <node name="image_view" pkg="image_view" type="image_view">
	<remap from="image" to="camera/rgb/image_raw" />
    </node>

    <node
        pkg="rosbag"
        name="record"
        type="record"
        args="--duration=30.0 -a -x /camera/(.*) -O $(find pytheas)/results/pytheas.bag"
        if="$(arg record)"
    />
</launch>